osu.Game/Graphics/Containers/Markdown/OsuMarkdownListItem.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using Markdig.Syntax;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Containers.Markdown;
+using osuTK;
+
+namespace osu.Game.Graphics.Containers.Markdown
+{
+    public class OsuMarkdownListItem : CompositeDrawable
+    {
+        private const float ordered_left_padding = 30;
+        private const float unordered_left_padding = 20;
+
+        private readonly int level;
+        private readonly int order;
+        private readonly bool isOrdered;
+
+        [Resolved]
+        private IMarkdownTextComponent parentTextComponent { get; set; }
+
+        public FillFlowContainer Content { get; }
+
+        public OsuMarkdownListItem(ListItemBlock listItemBlock, int level)
+        {
+            this.level = level;
+            order = listItemBlock.Order;
+            isOrdered = ((ListBlock)listItemBlock.Parent).IsOrdered;
```

Comment:
This should be split out into two components, rather than conditionaled multiple times in this file.

Reply:
I do this for simplicity to access `IsOrdered` field in `ListBlock`. Fortunately the `Parent` type of `ListItemBlock` is definitely `ListBlock`.

Reply:
The cast is not the problem, the fact that bulleted and numbered lists are one component is.

Reply:
Ah yeah, that could be done like that.

Or I can try approach like HTML `<ul>` and `<ol>` as `OsuMarkdown{Ordered,Unordered}List` that have `OsuMarkdownListItem` as its children.

Reply:
Sure, I guess. Whichever works best.